File: name-validation-3-nodejs/REST/nv3_response.js
/**
 * Validate name response component for NV3 API
 */
export class NV3Response {
    constructor(data = {}) {
        this.Status = data.status ?? null;
        this.NameResult = data.nameResult ? new NameResult(data.nameResult) : null;
        this.StatusDetail = data.statusDetail ? new StatusDetail(data.statusDetail) : null;
    }

    get IsSuccess() {
        return this.NameResult !== null;
    }
}

/**
 * NameResult component for NV3 API response
 */
export class NameResult {
    constructor(data = {}) {
        this.IsValidName = data.isValidName ?? null;
        this.Classification = data.classification ?? null;
        this.Confidence = data.confidence ?? null;
        this.TextIn = data.textIn ?? null;
        this.TextOut = data.textOut ?? null;
        this.ParsedName = data.parsedName ? new ParsedName(data.parsedName) : null;
        this.PossibleNames = Array.isArray(data.possibleNames)
            ? data.possibleNames.map(name => new NameInfo(name))
            : [];
        this.Notes = data.notes ?? null;
        this.Warnings = data.warnings ?? null;
        this.FirstNameFound = data.firstNameFound ?? null;
        this.IsCommonFirstName = data.isCommonFirstName ?? null;
        this.LastNameFound = data.lastNameFound ?? null;
        this.IsCommonLastName = data.isCommonLastName ?? null;
        this.SimilarFirstNames = data.similarFirstNames ?? null;
        this.SimilarLastNames = data.similarLastNames ?? null;
        this.RelatedNames = data.relatedNames ?? null;
    }
}

/**
 * ParsedName component for NV3 API response
 */
export class ParsedName {
    constructor(data = {}) {
        this.Prefix = data.prefix ?? null;
        this.First = data.first ?? null;
        this.Middle = data.middle ?? null;
        this.Last = data.last ?? null;
        this.Suffix = data.suffix ?? null;
    }
}

/**
 * NameInfo component for possibleNames
 */
export class NameInfo {
    constructor(data = {}) {
        this.Confidence = data.confidence ?? null;
        this.ParsedName = data.parsedName ? new ParsedName(data.parsedName) : null;
    }
}

/**
 * StatusDetail component for NV3 API errors
 */
export class StatusDetail {
    constructor(data = {}) {
        this.Message = data.message ?? null;
        this.Detail = data.detail ?? null;
    }
}

File: name-validation-3-python/REST/nv3_response.py
"""
Response classes for Name Validation 3 (NV3) API
"""
from dataclasses import dataclass
from typing import List, Optional

@dataclass
class StatusDetail:
    Message: Optional[str] = None
    Detail: Optional[str] = None

    def __str__(self):
        return f"StatusDetail(Message={self.Message}, Detail={self.Detail})"
    
@dataclass
class NV3Response:
    Status: Optional[str] = None
    NameResult: Optional[NameResult] = None
    StatusDetail: Optional[StatusDetail] = None

    def __str__(self):
        return (f"NV3Response(Status={self.Status}, NameResult={self.NameResult}, "
                f"StatusDetail={self.StatusDetail})")    

@dataclass
class NameResult:
    IsValidName: Optional[bool] = None
    Classification: Optional[str] = None
    Confidence: Optional[float] = None
    TextIn: Optional[str] = None
    TextOut: Optional[str] = None
    ParsedName: Optional[ParsedName] = None
    PossibleNames: Optional[List['PossibleName']] = None
    Notes: Optional[str] = None
    Warnings: Optional[str] = None
    FirstNameFound: Optional[bool] = None
    IsCommonFirstName: Optional[bool] = None
    LastNameFound: Optional[bool] = None
    IsCommonLastName: Optional[bool] = None
    SimilarFirstNames: Optional[List[str]] = None
    SimilarLastNames: Optional[List[str]] = None
    RelatedNames: Optional[List[str]] = None

    def __post_init__(self):
        if self.PossibleNames is None:
            self.PossibleNames = []
        if self.SimilarFirstNames is None:
            self.SimilarFirstNames = []
        if self.SimilarLastNames is None:
            self.SimilarLastNames = []
        if self.RelatedNames is None:
            self.RelatedNames = []

    def __str__(self):
        possible_names_str = "\n".join(str(name) for name in self.PossibleNames) if self.PossibleNames else "None"
        return (f"NameResult(Confidence={self.Confidence}, TextIn='{self.TextIn}', TextOut='{self.TextOut}', "
                f"PossibleNames=[{possible_names_str}], Notes='{self.Notes}', Warnings='{self.Warnings}', "
                f"FirstNameFound={self.FirstNameFound}, IsCommonFirstName={self.IsCommonFirstName}, "
                f"LastNameFound={self.LastNameFound}, IsCommonLastName={self.IsCommonLastName}, "
                f"SimilarFirstNames={self.SimilarFirstNames}, SimilarLastNames={self.SimilarLastNames}, "
                f"RelatedNames={self.RelatedNames})")
    
@dataclass
class ParsedName:
    Prefix: Optional[str] = None
    First: Optional[str] = None
    Middle: Optional[str] = None
    Last: Optional[str] = None
    Suffix: Optional[str] = None

    def __str__(self):
        return (f"ParsedName(Prefix='{self.Prefix}', First='{self.First}', Middle='{self.Middle}', "
                f"Last='{self.Last}', Suffix='{self.Suffix}')")
    
@dataclass
class PossibleName:
    Confidence: Optional[float] = None
    ParsedName: Optional[ParsedName] = None

    def __str__(self):
        return f"PossibleName(Confidence={self.Confidence}, ParsedName={self.ParsedName})"